app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadService.java

```diff
@@ -300,6 +298,11 @@ private void downloadBook(String url, int notificationID, LibraryNetworkEntity.B
         .distinctUntilChanged().doOnComplete(() -> updateDownloadFragmentComplete(notificationID))
         .subscribe(progress -> {
           if (progress == 100) {
+            java.util.List<File> chunks = FileUtils.getAllZimParts(book);
```

Comment:
I might be missing it but were is the second.part added?

Reply:
I've altered the logic to remove the `.part` only when the entire book has finished download. It's based on kelson's comment on #355. To keep the `.part` in the chunks and remove them only when the entire operation has finished. However, if you consider adding a second `.part` I could add and commit that today itself.

Reply:
I would rather have a second .part as chunk size could one day be variable.

Reply:
Right on it. Will update by tonight(IST).

Reply:
@mhutti1 I need help. I've finished coding but can't figure out a way to test this out.

Reply:
@sakchhams You can change the chunk size in one of the files to a much smaller size. Then you can test appropriately. Infact it might be worth adding some tests using this idea. (Might take some time though).

Reply:
@mhutti1 sorry for the delay I'm busy with my exams I'll get back on this ASAP.

Reply:
No problem. I have also had exams the last few weeks as you might be able to see from the pr backlog :P